apps/aevm/src/aevm_ae_primops.erl

```diff
@@ -0,0 +1,116 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2018, Aeternity Anstalt
+%%% @doc
+%%%     Handle interaction with the aeternity chain
+%%%     through calls to AEternity primitive operations at address 0.
+%%% @end
+%%% Created : 22 May 2018
+%%%-------------------------------------------------------------------
+
+-module(aevm_ae_primops).
+-export([call/3]).
+
+-include_lib("aebytecode/include/aeb_opcodes.hrl").
+
+-define(BASE_ADDRESS, 64). %% Byte offset for data
+-define(PRIMOP_BASE_ADDRESS, 96). %% Byte offset for data in primops
+-spec call( non_neg_integer(), binary(), aevm_eeevm_state:state()) ->
+                  {ok, binary(), non_neg_integer(), aevm_eeevm_state:state()}
+                      | {error, any()}.
+call(Value, Data, State) ->
+    %% TODO: use aeso_data:from_binary() (but it doesn't take a base address at the moment)
+    case Data of
+        <<?BASE_ADDRESS:256, ?PRIM_CALL_SPEND:256, Recipient:256>> ->
+            spend(Recipient, Value, State);
+	<<?PRIMOP_BASE_ADDRESS:256, _:256, Type:256, Argument/binary>> ->
```

Comment:
@UlfNorell This was not the encoding I expected.

Reply:
Matching on the pointers is not a good idea (`?PRIMOP_BASE_ADDRESS`). The encoding scheme doesn't uniquely determine the memory layout of the components. In this case the argument tuple happens to start at address 96. We really should parse the data.

Reply:
OK, thanks I’ll do that then.